pire/scanners/slow.h

```diff
@@ -249,39 +249,18 @@ class SlowScanner {
 		}
 	}
 
-	explicit SlowScanner(Fsm& fsm, size_t distance = 0, bool needActions = false, bool removeEpsilons = true)
+	explicit SlowScanner(Fsm& fsm, bool needActions = false, bool removeEpsilons = true)
 		: need_actions(needActions)
 	{
-		if (distance) {
-			fsm = CreateApproxFsm(fsm, distance);
-		}
-		if (removeEpsilons)
-			fsm.RemoveEpsilons();
-		fsm.Sparse(!removeEpsilons);
-
-		m.statesCount = fsm.Size();
-		m.lettersCount = fsm.Letters().Size();
-
-		m_vec.resize(m.statesCount * m.lettersCount);
-		if (need_actions)
-			m_actionsvec.resize(m.statesCount * m.lettersCount);
-		m_vecptr = &m_vec;
-		alloc(m_letters, MaxChar);
-		m_jumps = 0;
-		m_actions = 0;
-		m_jumpPos = 0;
-		alloc(m_finals, m.statesCount);
-
-		// Build letter translation table
-		Fill(m_letters, m_letters + sizeof(m_letters)/sizeof(*m_letters), 0);
-		for (auto&& letter : fsm.Letters())
-			for (auto&& character : letter.second.second)
-				m_letters[character] = letter.second.first;
-
-		m.start = fsm.Initial();
-		BuildScanner(fsm, *this);
+		BuildScanner(BuildInternalFsm(fsm, removeEpsilons), *this);
 	}
 
+	explicit SlowScanner(Fsm& fsm, size_t distance, bool needActions = false, bool removeEpsilons = true)
```

Comment:
IMHO
It would be better to make `distance` the last argument as Slava has suggested.
Different order of arguments is inconsistent and thus violates the principle of least astonishment ( https://en.wikipedia.org/wiki/Principle_of_least_astonishment ).
Scott Meyers mentions the same situation in the C library:  https://youtu.be/5tg1ONG18H8?t=2173 .
BTW, the talk is great  and I highly recommend it.